# Microbial-diversity-with-QIIME2

•	Microbial diversity in coconut leaves associated with the lethal yellowing disease symptoms in coastal Kenya

•	Qiime2 is one of the latest bioinformatics tools used in microbial diversity analysis.

•	For set up and installation of QIIME2 use the link: [Set Up And Installation (QIIME2)](https://github.com/Fatma366/16S-rRNA-End-to-End-Analysis#set-up-and-installation-qiime2)

•	For tutorials involving QIIME2 use the link: [QIIME2 Analysis](https://github.com/Fatma366/16S-rRNA-End-to-End-Analysis#qiime2-analysis)


•	The analysis involves the following steps: Included in the Microbial-Qiime2-analysis.sh and the last step in sample-classifier.sh

1.	Importing data to qiime2

2.	Demultiplexing allows you to determine how many sequences were obtained per sample

3.	Denoising, truncating and quality control using dada2

4.	Feature Table and Feature Data Summary

5.	Phylogenetic Tree Construction

6.	Visualization of the tree, though the tree will have features ID as the names, not so informative

7.	Alpha Rarefraction Plotting

8.	Alpha Diversity Analysis

9.	Beta Diversity Analysis

10.	Beta Group Significance Analysis (PERMANOVA)

11.	Taxonomic Analysis: Download the trained naive Bayes classifier for the v4 hypervariable region

12.	Predicting categorical sample data with machine learning classifiers

### How to Contribute
1. Clone the repository.
2. Create a new branch: `git checkout -b feature/your-feature-name`.
3. Commit your changes: `git commit -m 'Add some feature'`.
4. Push to the branch: `git push origin feature/your-feature-name`.
5. Submit a pull request.

### Code of Conduct
Please review and adhere to our [Code of Conduct](https://github.com/Fatma366/Identification-and-Prediction-using-ML/blob/main/CODE_OF_CONDUCT.md) to ensure a positive and inclusive community.

## License
This project is licensed under the MIT License. See the [LICENSE.md](https://github.com/Fatma366/Identification-and-Prediction-using-ML/blob/main/LICENSE.md) file for details.

